src/gsy_e/models/strategy/market_agents/two_sided_engine.py

```diff
@@ -47,6 +47,22 @@ def __repr__(self):
         return "<TwoSidedPayAsBidEngine [{s.owner.name}] {s.name} " \
                "{s.markets.source.time_slot:%H:%M}>".format(s=self)
 
+    def _update_requirements_prices(self, bid):
+        requirements = []
+        for requirement in bid.requirements or []:
+            updated_requirement = {**requirement}
+            if "price" in updated_requirement:
+                if "original_price" not in updated_requirement:
+                    updated_requirement["original_price"] = updated_requirement["price"]
+
+                energy = updated_requirement.get("energy") or bid.energy
+                updated_price = (self.markets.source.fee_class.update_forwarded_bid_with_fee(
+                    updated_requirement["price"] / energy,
+                    updated_requirement["original_price"] / energy)) * energy
```

Comment:
NIT: I found the bracket placing weird. As it works as you proposed, you can leave it if you like. 
```suggestion
                    updated_requirement["original_price"] / energy) * energy)
```